File: examples/csharp-vstest/steps/yahtzee.steps.cs
using System.Globalization;
using Varar;
using Varar.Core;
using static Varar.Example.StepHelpers;

namespace Varar.Example;

public static class YahtzeeSteps
{
    public static void Register(Steps s)
    {
        // Header-bound: dice and category are inputs, score is the computed column Varar checks.
        s.Sensor("Examples of dice, category and score", (state, row) =>
        {
            var m = SMap(row);
            var dice = AsStr(m["dice"]).Split(',').Select(d => int.Parse(d.Trim(), CultureInfo.InvariantCulture)).ToList();
            var category = AsStr(m["category"]);
            return VMap(("score", Value.Of(Yahtzee.Score(dice, category))));
        });
    }
}

File: examples/typescript-vitest/src/varar/yahtzee.steps.ts
import { steps } from '@varar/varar'
import { score } from '../src/yahtzee'

const { sensor } = steps()

sensor(
  'Examples of dice, category and score',
  (_state, row: { dice: string; category: string; score: string }) => {
    const dice = row.dice.split(',').map((d) => Number(d.trim()))
    return { score: score(dice, row.category) }
  },
)
